src/pages/course/[name]/problem/new.vue

```diff
@@ -63,10 +63,15 @@ async function submit() {
       })
     ).data;
 
-    // TODO: additional handling if testcase upload fail
     const testdataForm = new FormData();
     testdataForm.append("case", testdata.value);
-    await api.Problem.modifyTestdata(problemId, testdataForm);
+    try {
+      await api.Problem.modifyTestdata(problemId, testdataForm);
+    } catch (error) {
+      alert("Problem created, but testdata upload failed.");
+      router.push(`/course/${route.params.name}/problem/${problemId}/edit`);
+      throw error;
```

Comment:
Perhaps outside the scope of this issue. Could we add more detail to the UI about the error being responded to? What information needs to be displayed?

Reply:
Make sense, let me fix this

Reply:
fixed, take a look